app/controllers/announcements_controller.rb

```diff
@@ -0,0 +1,71 @@
+class AnnouncementsController < ApplicationController
+  before_action :set_announcement, only: [:edit, :update, :destroy]
+  before_action :new_announcement, only: [:new, :create, :index]
+
+  def show
+    redirect_to action: "index"
+  end
+
+  def index
+    @announcement = Announcement.joins(:user).active_announcements.not_expired.order(reveal_date: :desc)
+  end
+
+  def new
+    @announcement.reveal_date = Date.today
+    @announcement.expires_at = Date.today + 1.months
+  end
+
+  def create
+    @announcement = Announcement.new(announcement_params)
+    @announcement.user_id = current_user.id
+
+    if @announcement.save
+      redirect_to action: "index"
+    else
+      render "new"
+    end
+  end
+
+  def edit
+    @announcement = Announcement.find(params[:id])
+  end
+
+  def update
+    if @announcement.validate(params[:announcement])
+      @announcement.update(announcement_params)
+      flash[:success] = "Successfully modified and saved announcement."
+      redirect_to @announcement
+    else
+      flash[:danger] = "Could not modify announcement information."
+      render :edit
+    end
+  end
```

Comment:
change this to:
```ruby
def update
  @announcement.update(announcement_params)
  if @announcement.save
    flash[:success] = "Successfully modified and saved announcement."
    redirect_to action: "index"
  else
    flash[:danger] = "Could not modify announcement information."
    render :edit
  end
end
```
because the validate method is handled with the save call, the reason I use it in my code for `routes` and `assigned_routes` is because they use the form objects since they share the `route.rb` object. 
also you don't want to redirect to the individual announcement, because you don't have `show` pages setup for that.